### PR TITLE
Synthetic maintenance case

### PR DESCRIPTION
**Pull request by Mister-moon1**
# About the pull request
Adds a weak form of nano-paste, a tool able to remove shrapnel and a synthetic maintenance case as standard issue to working joe

# Explain why it's good for the game
The boot knife is planned for removal from working joe partially because of the fact it is a weapon, the addition of a tool that can remove shrapnel will allow working joes to combat any friendly fire cases without dying from moving.

This will be compacted into a small kit including nanopaste which will be issued to cupboards in a later PR, the intent is for any situation where a synthetic is dead that the kit will be used as a field bandage in a sense where the medic takes it and repairs them with it

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
add: adds a synthetic maintenance case
add: adds a shrapnel removal tool
add: adds a weak form of nano-paste
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/nanopaste.dm [hunk 1]
```diff
@@ -47,3 +47,51 @@
...
+	icon_state = "tube"
+
+	attack_speed = 3
+	amount = 10
+	max_amount = 10
+	w_class = SIZE_SMALL
+	stack_id = "nanopaste"
+	black_market_value = 10
+
+/obj/item/stack/nanopaste/weak/attack(mob/living/M as mob, mob/user as mob)
```
Comment (BeagleGaming1) [inline code/game/objects/items/stacks/nanopaste.dm:64]: single letter var

File: code/game/objects/items/stacks/nanopaste.dm [hunk 2]
```diff
@@ -47,3 +47,51 @@
...
+	if (!istype(M) || !istype(user))
+		return 0
+	if (isrobot(M)) //Repairing cyborgs
+		var/mob/living/silicon/robot/R = M
```
Comment (BeagleGaming1) [inline code/game/objects/items/stacks/nanopaste.dm:68]: single letter var

File: code/game/objects/items/stacks/nanopaste.dm [hunk 3]
```diff
@@ -47,3 +47,51 @@
...
+			R.apply_damage(-6, BURN)
+			R.updatehealth()
+			use(1)
+			user.visible_message(SPAN_NOTICE("\The [user] applied some [src] at [R]'s damaged areas."),\
+				SPAN_NOTICE("You apply some [src] at [R]'s damaged areas."))
+		else
+			to_chat(user, SPAN_NOTICE("All [R]'s systems are nominal."))
+
+	if (istype(M,/mob/living/carbon/human)) //Repairing robolimbs
+		var/mob/living/carbon/human/H = M
```
Comment (BeagleGaming1) [inline code/game/objects/items/stacks/nanopaste.dm:80]: single letter var

File: code/game/objects/items/stacks/nanopaste.dm [hunk 4]
```diff
@@ -47,3 +47,51 @@
...
+		if(isspeciessynth(H) && M == user && !H.allow_gun_usage)
+			to_chat(H, SPAN_WARNING("Your programming forbids you from self-repairing with \the [src]."))
+			return
+		var/obj/limb/S = H.get_limb(user.zone_selected)
```
Comment (BeagleGaming1) [inline code/game/objects/items/stacks/nanopaste.dm:84]: single letter var

**Review by BeagleGaming1 — CHANGES_REQUESTED** [review]

File: code/modules/gear_presets/synths.dm [hunk 5]
```diff
@@ -490,6 +490,7 @@
 	new_human.equip_to_slot_or_del(new /obj/item/storage/bag/trash(new_human), WEAR_L_HAND)
 	new_human.equip_to_slot_or_del(new /obj/item/device/lightreplacer(new_human.back), WEAR_IN_R_STORE)
 	new_human.equip_to_slot_or_del(new /obj/item/maintenance_jack(new_human), WEAR_J_STORE)
+	H.equip_to_slot_or_del(new /obj/item/storage/synthetic_maintenance_case/regular(H), WEAR_IN_BACK)
```
Comment (BeagleGaming1) [inline code/modules/gear_presets/synths.dm:493]:
```suggestion
	new_human.equip_to_slot_or_del(new /obj/item/storage/synthetic_maintenance_case/regular(new_human), WEAR_IN_BACK)
```

File: code/game/objects/items/storage/toolbox.dm [hunk 6]
```diff
@@ -81,3 +81,25 @@
 	new /obj/item/stack/cable_coil(src,30,color)
 	new /obj/item/tool/wirecutters(src)
 	new /obj/item/device/multitool(src)
+
+obj/item/storage/synthetic_maintenance_case
```
Comment (BeagleGaming1) [inline code/game/objects/items/storage/toolbox.dm:85]:
```suggestion
/obj/item/storage/synthetic_maintenance_case
```

File: code/game/objects/items/stacks/nanopaste.dm [hunk 7]
```diff
@@ -47,3 +47,44 @@
...
 				to_chat(user, SPAN_NOTICE("Nothing to fix here."))
 
+/obj/item/stack/nanopaste/weak
+	name = "low power nanopaste"
+	singular_name = "low power nanite swarm"
+	desc = "A cheap alternative to nanopaste. Fairly effective in repairing robotic machinery."
+
+	black_market_value = 10
+
+/obj/item/stack/nanopaste/weak/attack(mob/living/Mob as mob, mob/user as mob)
```
Comment (harryob) [inline code/game/objects/items/stacks/nanopaste.dm:57]: dont copy paste entire procs to change some number - make var/heal_amount on the parent type and adjust it on the subtype, and use heal_amount in the parent attack proc

Comment (harryob) [reply code/game/objects/items/stacks/nanopaste.dm:57]: also this isn’t snakecase

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/game/objects/items/stacks/nanopaste.dm (hunk 7 above)
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:57]: `as thing` does nothing; args/vars should be lower case

File: code/game/objects/items/stacks/nanopaste.dm [hunk 8]
```diff
@@ -47,3 +47,44 @@
...
+	if (!istype(Mob) || !istype(user))
+		return 0
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:59]: use bool defines

File: code/game/objects/items/stacks/nanopaste.dm [hunk 9]
```diff
@@ -47,3 +47,44 @@
...
+	if (isrobot(Mob)) //Repairing cyborgs
+		var/mob/living/silicon/robot/synth = Mob
+		if (synth.getBruteLoss() || synth.getFireLoss() )
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:62]:
```suggestion
		if (synth.getBruteLoss() || synth.getFireLoss())
```

File: code/game/objects/items/stacks/nanopaste.dm [hunk 10]
```diff
@@ -47,3 +47,44 @@
...
+			synth.apply_damage(-6, BRUTE)
+			synth.apply_damage(-6, BURN)
+			synth.updatehealth()
+			use(1)
+			user.visible_message(SPAN_NOTICE("\The [user] applied some [src] at [synth]'s damaged areas."),\
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:67]:
```suggestion
			user.visible_message(SPAN_NOTICE("[user] applies some [src] on [synth]'s damaged areas."),\
```

File: code/game/objects/items/stacks/nanopaste.dm [hunk 11]
```diff
@@ -47,3 +47,44 @@
...
+				SPAN_NOTICE("You apply some [src] at [synth]'s damaged areas."))
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:68]:
```suggestion
				SPAN_NOTICE("You apply some [src] on [synth]'s damaged areas."))
```

File: code/game/objects/items/stacks/nanopaste.dm [hunk 12]
```diff
@@ -47,3 +47,44 @@
...
+		else
+			to_chat(user, SPAN_NOTICE("All [synth]'s systems are nominal."))
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:70]:
```suggestion
			to_chat(user, SPAN_NOTICE("All of [synth]'s systems are fully working."))
```

File: code/game/objects/items/stacks/nanopaste.dm [hunk 13]
```diff
@@ -47,3 +47,44 @@
...
+
+	if (istype(Mob,/mob/living/carbon/human)) //Repairing robolimbs
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:72]: ishuman macro

File: code/game/objects/items/stacks/nanopaste.dm [hunk 14]
```diff
@@ -47,3 +47,44 @@
...
+		var/mob/living/carbon/human/H = Mob
+		if(isspeciessynth(H) && Mob == user && !H.allow_gun_usage)
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:74]: `!H.allow_gun_usage`
why?

File: code/game/objects/items/stacks/nanopaste.dm [hunk 15]
```diff
@@ -47,3 +47,44 @@
...
+			synth.apply_damage(-6, BURN)
+			synth.updatehealth()
+			use(1)
+			user.visible_message(SPAN_NOTICE("\The [user] applied some [src] at [synth]'s damaged areas."),\
+				SPAN_NOTICE("You apply some [src] at [synth]'s damaged areas."))
+		else
+			to_chat(user, SPAN_NOTICE("All [synth]'s systems are nominal."))
+
+	if (istype(Mob,/mob/living/carbon/human)) //Repairing robolimbs
+		var/mob/living/carbon/human/H = Mob
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:73]: one letter variable

File: code/game/objects/items/stacks/nanopaste.dm [hunk 16]
```diff
@@ -47,3 +47,44 @@
...
+		if(isspeciessynth(H) && Mob == user && !H.allow_gun_usage)
+			to_chat(H, SPAN_WARNING("Your programming forbids you from self-repairing with \the [src]."))
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:75]:
```suggestion
			to_chat(H, SPAN_WARNING("Your programming forbids you from self-repairing with [src]."))
```

File: code/game/objects/items/stacks/nanopaste.dm [hunk 17]
```diff
@@ -47,3 +47,44 @@
...
+			return
+		var/obj/limb/S = H.get_limb(user.zone_selected)
+
+		if (S && (S.status & (LIMB_ROBOT|LIMB_SYNTHSKIN)))
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:79]:
```suggestion
		if(S?.status & (LIMB_ROBOT|LIMB_SYNTHSKIN))
```

File: code/game/objects/items/stacks/nanopaste.dm [hunk 18]
```diff
@@ -47,3 +47,44 @@
...
+			if(S.get_damage())
+				S.heal_damage(6, 6, robo_repair = 1)
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:81]:
```suggestion
				S.heal_damage(6, 6, robo_repair = TRUE)
```

File: code/game/objects/items/stacks/nanopaste.dm [hunk 19]
```diff
@@ -47,3 +47,44 @@
...
+				H.pain.recalculate_pain()
+				H.updatehealth()
+				use(1)
+				var/others_msg = "\The [user] applies some nanite paste at[user != Mob ? " \the [Mob]'s" : " \the"] [S.display_name] with \the [src]."
+				var/user_msg = "You apply some nanite paste at [user == Mob ? "your" : "[Mob]'s"] [S.display_name]."
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:86]: why are these variables

File: code/game/objects/items/stacks/nanopaste.dm [hunk 20]
```diff
@@ -47,3 +47,44 @@
...
+				user.visible_message(SPAN_NOTICE("[others_msg]"),\
+					SPAN_NOTICE("[user_msg]"))
+			else
+				to_chat(user, SPAN_NOTICE("Nothing to fix here."))
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:90]:
```suggestion
				to_chat(user, SPAN_NOTICE("There is nothing to fix here!"))
```

File: code/game/objects/items/stacks/nanopaste.dm [hunk 21]
```diff
@@ -47,3 +47,44 @@
...
+			return
+		var/obj/limb/S = H.get_limb(user.zone_selected)
+
+		if (S && (S.status & (LIMB_ROBOT|LIMB_SYNTHSKIN)))
+			if(S.get_damage())
+				S.heal_damage(6, 6, robo_repair = 1)
+				H.pain.recalculate_pain()
+				H.updatehealth()
+				use(1)
+				var/others_msg = "\The [user] applies some nanite paste at[user != Mob ? " \the [Mob]'s" : " \the"] [S.display_name] with \the [src]."
```
Comment (Zonespace27) [inline code/game/objects/items/stacks/nanopaste.dm:85]:
```suggestion
				var/others_msg = "[user] applies some [src] on [H]'s [S.display_name]."
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by Mister-moon1** [comment]
After consideration and the fact that 2 nanopaste rolls were going to be in the case, i have lowered the amount to 1 roll and switched it out for normal nanopaste since it will heal for similar amounts

File: code/game/objects/items/storage/toolbox.dm [hunk 22]
```diff
@@ -81,3 +81,23 @@
...
+	w_class = SIZE_LARGE
+	matter = list("plastic" = 1000)
+	can_hold = list(
+		/obj/item/tool/surgery/shrapnel_remover,
+		/obj/item/stack/nanopaste,
+	)
+
+/obj/item/storage/synthetic_maintenance_case/regular
+
+/obj/item/storage/synthetic_maintenance_case/regular/fill_preset_inventory()
```
Comment (Zonespace27) [inline code/game/objects/items/storage/toolbox.dm:101]: At this point just remove the subtype

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**